py/torch_tensorrt/dynamo/conversion/impl/select.py

```diff
@@ -61,3 +69,271 @@ def select(
     if len(out.shape) != 1:
         layer = network.add_shuffle(out)
     return layer.get_output(0)
+
+
+def index(
+    network: TRTNetwork,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input: TRTTensor,
+    index: Union[TRTTensor, Sequence[TRTTensor]],
+) -> TRTTensor:
+    adv_indx_indices = []
+    tensor_indices = []
+    # _LOGGER.debug(f"The index shape is {index.shape}")
+    # check if the input is dynamic
+    dynamic_shape = has_dynamic_shape(input.shape)
+
+    # here we need to check if all the index are broadcastable
+    # if no, then we need to broadcast
+
+    last_index = None
+    for i, ind in enumerate(index):
+        if ind is not None:
+            _LOGGER.debug(f"Shape of {i} index is {ind.shape}")
+            adv_indx_indices.append(i)
+            # torch.nn.parameter.Parameter=> torch.Tensor
+            ind = get_trt_tensor(network, ind, name + f"_parameter_to_fp32_tensor_{i}")
+            if last_index is not None:
+                assert broadcastable(ind, last_index), "The indices should be broadcastable!"
+            last_index = ind
+            tensor_indices.append(ind)
+
+    if not tensor_indices:
+        identity_layer = network.add_identity(input)
+        identity_layer.set_output_type(0, trt.int32)
+        set_layer_name(identity_layer, target, name + "_index_identity", source_ir)
+        return identity_layer.get_output(0)
+    elif len(tensor_indices) == 1:
+        # This case works
+        indices_tensor = tensor_indices[0]
+        index = adv_indx_indices[0]
+        _LOGGER.debug(f"The advanced index indices is {adv_indx_indices}")
+        gather_layer = network.add_gather(input, indices_tensor, index)
+        set_layer_name(gather_layer, target, name + "_index_gather", source_ir)
+        return gather_layer.get_output(0)
+    else:
+        input_shape = input.shape
+        _LOGGER.debug(f"The input shape is {input.shape}")
+        if dynamic_shape:
+            input_shape = get_shape_with_dynamic_shape(
+                network, target, source_ir, name, input_shape, input
+            )
+        rank = len(input_shape)
+        adv_indx_count = len(adv_indx_indices)
+        dim_tensor_list = []
+
+        for i in range(rank):
+            dim = input_shape[i]
+            dim_tensor = get_trt_tensor(network, dim, name + f"individual_dim_{i}")
```

Comment:
Consider adding an underscore to the string after the name for better readability. Also applies to any of the below instances of `name + *`